src/server/helpers/auth.js

```diff
@@ -65,22 +62,25 @@ auth.init = (app) => {
 				passReqToCallback: true
 			}, config.musicbrainz
 		),
-		(req, accessToken, refreshToken, profile, done) => {
-			if (req.user) {
-				// Logged in, associate
-				return linkMBAccount(req.user, profile)
-					.then((linkedUser) => done(null, linkedUser.toJSON()));
-			}
+		async (req, accessToken, refreshToken, profile, done) => {
+			try {
+				if (req.user) {
+					const linkedUser = await _linkMBAccount(req.user, profile);
+
+					// Logged in, associate
+					return done(null, linkedUser.toJSON());
+				}
+
+				// Not logged in, authenticate
+				const fetchedUser = await _getAccountByMBUserId(profile);
+
+				await _updateCachedMBName(fetchedUser, profile);
```

Comment:
Unsure here; does the save mutate `fetchedUser` in any way, or are we fine holding onto the previous reference?